gdml-generator/src/SimpleLZ.cc

```diff
@@ -0,0 +1,231 @@
+ * The PMTs are spaced 1 mm apart, and sized such that the array is inscribed
+ * within the world cyclinder. These notional geometries are useful for
+ * performance studies.
+ */
+G4VPhysicalVolume* SimpleLZ::Construct()
+{
+    // Materials
+    auto nist = G4NistManager::Instance();
+    auto pmt_mat = nist->FindOrBuildMaterial("G4_Au");
+    auto world_mat = nist->FindOrBuildMaterial("G4_Galactic");
+    world_mat->SetName("vacuum");
+
+    // World Cylinder
+    G4Tubs* world_cylinder = new G4Tubs(
+        "world_cylinder", 0, 780.288 * mm, 123 * mm, 0 * deg, 360 * deg);
+    auto const world_lv
+        = new G4LogicalVolume(world_cylinder, world_mat, "world");
+    auto const world_pv = new G4PVPlacement(
+        nullptr, G4ThreeVector(), world_lv, "world_pv", nullptr, false, 0, false);
+
+    if (sqrt_num_pmts_ == 0)
+    {
+        // Simple LZ model
+        for (auto i = 0; i < 253; ++i)
+        {
+            G4Tubs* pmt
+                = new G4Tubs("pmt", 0, 39 * mm, 123 * mm, 0 * deg, 360 * deg);
+            std::string pmt_name = "pmt_" + std::to_string(i);
+            auto const pmt_lv = new G4LogicalVolume(pmt, pmt_mat, pmt_name);
+            new G4PVPlacement(nullptr,
+                              G4ThreeVector({pmt_x[i] * mm, pmt_y[i] * mm, 0}),
+                              pmt_lv,
+                              "pmt_pv",
+                              world_lv,
+                              false,
+                              0,
+                              false);
+        }
+    }
+    else
+    {
+        // Notional model
+        double a = 780.288 * std::sqrt(2) * mm;
+        double pmt_pitch = a / (sqrt_num_pmts_);
+        double pmt_radius = (pmt_pitch - 1 * mm) / 2;
+        double offset = a / 2 - pmt_pitch / 2;
+
+        for (auto i = 0; i < sqrt_num_pmts_; ++i)
+        {
+            for (auto j = 0; j < sqrt_num_pmts_; ++j)
+            {
+                G4Tubs* pmt = new G4Tubs(
+                    "pmt", 0, pmt_radius, 123 * mm, 0 * deg, 360 * deg);
+                std::string pmt_name = "pmt_" + std::to_string(i);
+                auto const pmt_lv = new G4LogicalVolume(pmt, pmt_mat, pmt_name);
+                new G4PVPlacement(nullptr,
+                                  G4ThreeVector({pmt_pitch * i - offset,
+                                                 pmt_pitch * j - offset,
+                                                 0 * mm}),
+                                  pmt_lv,
+                                  "pmt_pv",
+                                  world_lv,
+                                  false,
+                                  0,
+                                  false);
+            }
+        }
+    }
+
+    return world_pv;
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Set sensitive detectors.
+ */
+void SimpleLZ::ConstructSDandField()
+{
+    auto slab_sb = new SensitiveDetector("cylinder_sd");
```

Comment:
If you do not need SDs, then this should be an empty call.

If we'd like to score hits in the PMTs, then we can create a `SensitiveDetector("pmt_sd")` and assign it to the pmts logical volume.

If we go for one logical volume, multiple placements, then these three lines take care of everything.
If we go for one logical volume per placement, then we need a for loop and create one `pmt_sd_[i]`, each assigned to `pmt_i` (the name you gave to the logical volume in the for loop)